core/src/main/java/jenkins/model/FeatureSwitchConfiguration.java

```diff
@@ -0,0 +1,165 @@
+
+    public EnumSet<Feature> getDisabledSet() {
+        return disabledSet;
+    }
+
+    /**
+     * Update the disabled feature list
+     * used by {@link #configure(StaplerRequest req, JSONObject json)}.
+     *
+     * @param disabledSet
+     */
+    public void setDisabledSet(EnumSet<Feature> disabledSet) {
+        this.disabledSet = disabledSet;
+        this.disabledFeatureStr.clear();
+        for (Feature feature : disabledSet) {
+            disabledFeatureStr.add(feature.name());
+        }
+        save();
+    }
+
+    //Allow enum to be removed in future release, avoid serialization/deserialization
+    transient EnumSet<Feature> disabledSet = EnumSet.noneOf(Feature.class);
+
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+        req.bindJSON(this, json);
+        return true;
+    }
+
+    public FeatureSwitchConfiguration() {
+        //Load configure data from disk
+        load();
+        updateFeatureSet();
+    }
+
+    public static FeatureSwitchConfiguration get() {
+        return GlobalConfiguration.all().get(FeatureSwitchConfiguration.class);
+    }
+
+    /**
+     * Convert string to enum, removed feature(no matching enum item)
+     * will be discarded
+     */
+    private void updateFeatureSet() {
+        for (Feature feature : Feature.values()) {
+            if (disabledFeatureStr.contains(feature.name())) {
+                disabledSet.add(feature);
+            }
+        }
+    }
+
+    /**
+     * Check the feature is enabled or not
+     *
+     * @param feature
+     * @return a boolean flag indicating the feature disabled or not
+     */
+    public static boolean isDisabled(Feature feature) {
+        return get().getDisabledSet().contains(feature);
+    }
+
+    public enum Feature {
+        /**
+         * User can leverage the feature to fast node remove
+         * <pre>{@code
+         *  SomeRetentionStrategy extends RetentionStrategy<SlaveComputer> implements ExecutorListener {
+         *     @Override
+         *     public void taskCompleted(Executor executor, Queue.Task task, long durationMS) {
+         *         final SomeCloudSlave slaveNode=(SomeCloudSlave)executor.getOwner().getNode();
+         *         if(slaveNode!=null && slaveNode.isOnetimeUse()){
+         *             slaveNode.setAcceptingTasks(false);
+         *             Computer.threadPoolForRemoting.submit(new Runnable() {
+         *                 public void run() {
+         *                     //some cleanup code here
+         *                     //remove node after cleanup
+         *                     Jenkins.getInstance().removeNode(slaveNode);
+         *                 }
+         *             });
+         *         }
+         *     }
+         *  }
+         * }</pre>
+         */
+        DEFER_EXECUTOR_CREATION(
+                "Defer executor creation for nodes which are not accepting task"
```

Comment:
there are non english speaking users, so this should be a Messages key or the like so it can be replaced with French etc.